model/common/src/icon4py/model/common/utils/gt4py_field_allocation.py

```diff
@@ -5,13 +5,49 @@
 #
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
+import logging as log
 from typing import Optional
 
+import gt4py._core.definitions as gt_core_defs
 import gt4py.next as gtx
 from gt4py.next import backend
 
-from icon4py.model.common import type_alias as ta
-from icon4py.model.common.settings import xp
+from icon4py.model.common import dimension, type_alias as ta
+
+
+def is_cupy_device(backend: backend.Backend) -> bool:
+    cuda_device_types = (
```

Comment:
Can we pull this tuple outside of this function and declare it as `_cuda_device_types`?

Reply:
yes.